.github/workflows/main.yml

```diff
@@ -0,0 +1,236 @@
+          node-version: ${{matrix.node}}
+
+      - uses: actions/setup-python@v5
+        with:
+          # More recent versions of python require more recent node-gyp
+          # See https://github.com/nodejs/node-gyp/issues/2869
+          python-version: "3.11"
+
+      - name: Install && Build TS + bindings
+        run: yarn
+
+      - name: Run Unit Tests
+        run: yarn test:unit
+
+      - name: Run Memory Tests
+        run: yarn test:memory
+
+      # Attempt to pull spec-tests from cache
+      - name: Restore spec tests cache
+        id: cache-restore
+        if: matrix.os != 'windows-latest'
+        uses: actions/cache/restore@v4
+        with:
+          path: spec-tests
+          key: spec-test-data-${{ hashFiles('test/spec/specTestVersioning.ts') }}
+
+      # Download spec tests if not persisted
+      - name: Download spec tests
+        if: matrix.os != 'windows-latest'
+        run: yarn download-spec-tests
+
+        # Cache Spec Tests if not already cached
+      - name: Save spec tests to cache
+        if: matrix.os != 'windows-latest' && steps.cache-restore.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v4
+        with:
+          path: spec-tests
+          key: spec-test-data-${{ hashFiles('test/spec/specTestVersioning.ts') }}
+
+      - name: Run spec tests
+        if: matrix.os != 'windows-latest'
+        run: yarn test:spec
+
+      - name: Upload binding.node
+        uses: actions/upload-artifact@v4
+        if: github.repository_owner == 'chainsafe' && github.event_name != 'pull_request'
+        with:
+          name: binding.node
+          path: prebuild/*.node
+          if-no-files-found: error
+
+  # Required as current base images do not support arm64
+  # https://resources.github.com/devops/accelerate-your-cicd-with-arm-and-gpu-runners-in-github-actions/
+  # https://github.com/actions/runner-images/issues/5631
+  build-arm:
+    strategy:
+      fail-fast: false
+      matrix:
+        node:
+          - 16
+          - 18
+          - 20
+          - 21
+    runs-on: ubuntu-22.04
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      # Download spec tests with cache
+      - name: Restore spec tests cache
+        uses: actions/cache@v4
+        with:
+          path: spec-tests
+          key: spec-test-data-${{ hashFiles('test/spec/specTestVersioning.ts') }}
+
+      # https://github.com/docker/setup-qemu-action
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      # https://github.com/docker/setup-buildx-action
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build arm64 Binding, Run Tests and Copy Build Artifact to Host
+        run: |
+          docker buildx build --build-arg NODE_VERSION=${{matrix.node}} --platform linux/arm64 -t blst-ts-test-armbuild:${{matrix.node}} --load --progress=plain .
```

Comment:
should this `blst-ts-test-armbuild` be renamed to `blst-ts-tmp-armbuild`?

Reply:
Sure! Updated to `blst-ts-temp-armbuild` (im neurotic and like the whole words).